jobs/sql/extractions/hiv_monthly_reporting.sql

```diff
@@ -0,0 +1,88 @@
+SET sql_safe_updates = 0;
+
+SET @hiv_program_id = (SELECT program_id FROM program WHERE retired = 0 AND uuid = 'b1cb1fc1-5190-4f7a-af08-48870975dafc');
+select name into @HIV_Intake_Name from encounter_type where uuid = 'c31d306a-40c4-11e7-a919-92ebcb67fe33' ;
+select name into @HIV_Followup_Name from encounter_type where uuid = 'c31d3312-40c4-11e7-a919-92ebcb67fe33' ;
+select name into @HIV_Dispensing_Name from encounter_type where uuid = 'cc1720c9-3e4c-4fa8-a7ec-40eeaad1958c' ;
+
+DROP TEMPORARY TABLE IF EXISTS temp_hiv_patient_programs;
+CREATE TEMPORARY TABLE temp_hiv_patient_programs
+SELECT patient_id, patient_program_id, date_enrolled, date_completed
+FROM patient_program WHERE voided=0 AND program_id = @hiv_program_id;
+
+
+DROP TEMPORARY TABLE IF EXISTS temp_eom_appts;
+CREATE TEMPORARY TABLE temp_eom_appts
+(
+    patient_id                  INT(11),
+    patient_program_id			INT(11),
+    date_enrolled				DATETIME,
+    date_completed				DATETIME,
+    reporting_date				DATE,
+	  latest_HIV_note_encounter_id	INT(11),
+    latest_HIV_visit_date		DATETIME,
+    latest_expected_HIV_visit_date	DATETIME,
+	  HIV_visit_days_late			INT,
+	  latest_dispensing_encounter_id	INT(11),
+    latest_dispensing_date		DATETIME,
+    latest_expected_dispensing_date	DATETIME,
+	  dispensing_days_late			INT
+);
+
+create index eom_patient on temp_eom_appts(patient_id);
+
+call load_end_of_month_dates;
+
+-- insert end of month date rows for each patient for when they are active 
+insert into temp_eom_appts (patient_id,patient_program_id,date_enrolled, date_completed, reporting_date)
+select * from temp_hiv_patient_programs t
+inner join END_OF_MONTH_DATES e 
+	on e.reporting_date >= last_day(t.date_enrolled)  
+	and (last_day(t.date_completed) >= e.reporting_date or t.date_completed is null)
+	and YEAR(e.reporting_date) >= 2020 -- << load rows only for year 2020 and on 
```

Comment:
Alternatively, you could load end_of_month_dates with the dates you care about reporting on, or turn the load_end_of_month_dates into a procedure that takes in a start_date, end_date and creates a temporary table with the end of month dates in this range, and then you just use that here.